internal/verifier/recheck.go

```diff
@@ -63,38 +84,78 @@ func (verifier *Verifier) insertRecheckDocs(
 
 	generation, _ := verifier.getGenerationWhileLocked()
 
-	models := []mongo.WriteModel{}
-	for i, documentID := range documentIDs {
-		pk := RecheckPrimaryKey{
-			Generation:     generation,
-			DatabaseName:   dbNames[i],
-			CollectionName: collNames[i],
-			DocumentID:     documentID,
-		}
+	docIDIndexes := lo.Range(len(documentIDs))
+	indexesPerThread := splitToChunks(docIDIndexes, recheckInserterThreadsSoftMax)
 
-		// The filter must exclude DataSize; otherwise, if a failed comparison
-		// and a change event happen on the same document for the same
-		// generation, the 2nd insert will fail because a) its filter won’t
-		// match anything, and b) it’ll try to insert a new document with the
-		// same _id as the one that the 1st insert already created.
-		filterDoc := bson.D{{"_id", pk}}
+	eg, groupCtx := errgroup.WithContext(ctx)
 
-		recheckDoc := RecheckDoc{
-			PrimaryKey: pk,
-			DataSize:   dataSizes[i],
-		}
+	for _, curThreadIndexes := range indexesPerThread {
+		curThreadIndexes := curThreadIndexes
```

Comment:
Why do we redefine the var?

Reply:
The linter flagged it as one of those cases where you’re supposed to redefine the var to prevent loop-variable-in-goroutine problems.